newIDE/app/src/AssetStore/index.js

```diff
@@ -374,13 +398,24 @@ export const AssetStore = React.forwardRef<Props, AssetStoreInterface>(
                     <SearchBar
                       placeholder={t`Search assets`}
                       value={searchText}
-                      onChange={setSearchText}
-                      onRequestSearch={() => {
-                        // Clear the history
-                        navigationState.activateTextualSearch();
-                        navigationState.clearHistory();
-                        setIsFiltersPanelOpen(true);
-                      }}
+                      onChange={
+                        isOnShowingSearchResultPage
+                          ? // An existing search is already being done: just update the
+                            // search text and the asset store will update the search results.
+                            setSearchText
+                          : (newValue: string) => {
+                              setSearchText(newValue);
+
+                              // A new search is being initiated: navigate to the search page,
+                              // and clear the history as a new search was launched.
+                              if (!!newValue) {
+                                navigationState.clearHistory();
+                                navigationState.openSearchPage();
+                                setIsFiltersPanelOpen(true);
+                              }
+                            }
+                      }
+                      onRequestSearch={() => {}}
```

Comment:
should we still keep this function and find a way to call the `changeValueImmediately()` from within the searchbar?
That it would feel really snappy when pressing enter.
(not sure it's worth it tbh)

Reply:
I think this would be possible... but conscious of time I won't do it for now as pressing Enter is almost indistinguishable  from waiting a few more milliseconds after pressing it even if it does nothing ;) 
But agree we could maybe at some point rework this.

Note that onRequestSearch is still worth having, not here, but in other components to select/enter in the first element (it should probably be renamed "onValidate" or something like this, but also not doing it in this PR).